ietf-layer0-types-ext@2021-02-04.yang

```diff
@@ -0,0 +1,813 @@
+    }
+  } // grouping common-explicit-mode    
+
+  grouping common-organizational-explicit-mode {
+    description "Common capability attributes limit range 
+    in case of operational mode and explicit mode. 
+    These attributes are supported separately in 
+    case of application codes";
+
+/* transmitter tuning range (f_tx-min, f_tx-max) */
+
+    leaf min-central-frequency {
+      type frequency-thz;
+      config false;
+      description
+        "This parameter indicates the minimum frequency for
+         this transceiver.";
+    }
+    leaf max-central-frequency {
+      type frequency-thz;
+      config false;
+      description
+        "This parameter indicates the maximum frequency
+         for this transceiver.";
+    }
+
+/* transmitter-tunability-grid */
+
+    leaf minimum-channel-spacing {
+      type frequency-ghz;
+      config false;
+      description
+        "This paramenter represents the minumum difference in
+         frequency between two adjacent channels
+         (ref. G.698.2 sec.7.1.1).
+         This free value is to permit OTSi's central frequency not
+         to stay on the G.694.1 grid.";
+    } 
+
+/* supported transmitter power range [p_tx-min, p_tx_max] */
+
+    leaf tx-channel-power-min {
+      type dbm-t;
+      config false;
+      description "The minimum output power of this interface";
+    }
+    leaf tx-channel-power-max {
+      type dbm-t;
+      config false;
+      description "The maximum output power of this interface";
+    } 
+
+/* supported receiver power range [p_rx-min, p_rx_max] */ 
+
+     leaf rx-channel-power-min {
+      type dbm-t;
+      config false;
+      description "The minimum input power of this interface";
+    }
+    leaf rx-channel-power-max {
+      type dbm-t;
+      config false;
+      description "The maximum input power of this interface";
+    }
+
+    leaf rx-total-power-max {
+      type dbm-t;
+      config false;
+      description "Maximum rx optical power for
+      all the channels";
+    }      
+
+  } // grouping common-organizational-explicit-mode
+
+/* This grouping represent the list of configured parameters */
+/* values independent of operational mode */
+
+  grouping common-transceiver-configured-param {
+    description "Capability of an optical transceiver";
+
+    leaf OTSi-carrier-frequency { 
+        type frequency-thz;
+         description
+           "OTSi carrier frequency, equivalent to the
+           actual configured transmitter frequency";
+    }
+    
+
+    leaf tx-channel-power {
+        type dbm-t;
+        description "The current channel transmit power";
+    }
+
+    leaf rx-channel-power {
+        type dbm-t;
+        config false;
+        description "The current channel received power ";
+    }
+
+    leaf rx-total-power {
+        type dbm-t;
+        config false;
+        description "Current total received power";
+    }
+
+  } // grouping for configured attributes out of mode
+
+
+} // end of module
```

Comment:
This comment needs to be removed or yanglint will throw error:
err : Trailing garbage "// end of modul..." after module, expected end-of-input. (Line number 813.)
err : Loading "ietf-layer0-types-ext" module failed.